File: 0705-design-hashset/0705-design-hashset.py
class MyHashSet:

    def __init__(self):
        self.Set = []
        self.log = {}
        

    def add(self, key: int) -> None:
        if self.log.get(key) is None:
            self.Set.append(key)
            self.log[key] = len(self.Set) -1
        

    def remove(self, key: int) -> None:
        if self.log.get(key) is not None:
            self.Set[self.log[key]] = -1
            del self.log[key]
        

    def contains(self, key: int) -> bool:
        return self.log.get(key) is not None
        


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)

File: 0705-design-hashset/0705-design-hashset.cpp
class MyHashSet {
public:
    int32_t arrSize;
    vector<bool> storage;
    MyHashSet() : arrSize(pow(10, 6) + 1), storage(arrSize, false) {}
    
    void add(int key) {
        storage[key] = true;
    }
    
    void remove(int key) {
        storage[key] = false;
    }
    
    bool contains(int key) { 
        return storage[key];
    }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */